phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRowTimestampFunctionIT.java

```diff
@@ -191,4 +225,130 @@ public void testRowTimestampColumn() throws Exception {
+            String sql = "SELECT PHOENIX_ROW_TIMESTAMP(), KV2 FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() < PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                int actualCount = 0;
+                while(rs.next()) {
+                    assertTrue(rs.getDate(1).before(new Date(rowTimestamp)));
+                    rs.getString(2);
+                    assertFalse(rs.wasNull());
+                    actualCount++;
+                }
+                assertEquals(NUM_ROWS, actualCount);
+                rs.close();
+            }
+        }
+
+    }
+
+    @Test
+    // case: All rows selected should have the phoenix_row_timestamp() > date column
+    // Since we used a past date for column PK2
+    public void testRowTimestampFunctionOnlyWithGreaterThanPredicate() throws Exception {
+        long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis() - TS_OFFSET;
+        String tableName = createTestData(rowTimestamp, NUM_ROWS);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String sql = "SELECT PHOENIX_ROW_TIMESTAMP() FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() > PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                int actualCount = 0;
+                while(rs.next()) {
+                    assertTrue(rs.getDate(1).after(new Date(rowTimestamp)));
+                    actualCount++;
+                }
+                assertEquals(NUM_ROWS, actualCount);
+                rs.close();
+            }
+        }
+    }
+
+    @Test
+    // case: All rows selected should have the phoenix_row_timestamp() > date column
+    // Since we used a past date for column PK2
+    // Additional columns should return non null values.
+    public void testRowTimestampFunctionAndAdditionalColsWithGreaterThanPredicate() throws Exception {
+        long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis() - TS_OFFSET;
+        String tableName = createTestData(rowTimestamp, NUM_ROWS);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String sql = "SELECT PHOENIX_ROW_TIMESTAMP(), KV1 FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() > PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                int actualCount = 0;
+                while(rs.next()) {
+                    assertTrue(rs.getDate(1).after(new Date(rowTimestamp)));
+                    rs.getString(2);
+                    assertFalse(rs.wasNull());
+                    actualCount++;
+                }
+                assertEquals(NUM_ROWS, actualCount);
+                rs.close();
+            }
+        }
+    }
+
```

Comment:
better to support test for `SELECT COLUMN_ONLY WHERE PHOENIX_ROW_TIMESTAMP()` predicate.